engines/buried/inventory_info.cpp

```diff
@@ -0,0 +1,308 @@
+		int numLines = _viewLineCount[_curView];
+		uint32 boxColor = _vm->_gfx->getColor(255, 0, 0);
+		Common::Rect box(1, (187 / numLines) * _curLineIndex, 430, (187 / numLines) * (_curLineIndex + 1) - 1);
+		box.translate(absoluteRect.left, absoluteRect.top);
+		_vm->_gfx->getScreen()->frameRect(box, boxColor);
+	}
+}
+
+void BurnedLetterViewWindow::onLButtonUp(const Common::Point &point, uint flags) {
+	if (_top.contains(point) && _curView > 0) {
+		_curView--;
+		_curLineIndex = -1;
+
+		int offset = _vm->_gfx->computeVPushOffset(_vm->getTransitionSpeed());
+
+		// Only draw if transitions are enabled
+		if (offset != 189) {
+			TempCursorChange cursorChange(kCursorWait);
+
+			Graphics::Surface *newFrame = _stillFrames->getFrameCopy(_curView);
+		
+			for (int i = 0; i < 189; i += offset) {
+				_preBuffer->move(0, offset, _preBuffer->h);
+
+				for (int j = 0; j < offset; j++)
+					memcpy(_preBuffer->getBasePtr(0, j), newFrame->getBasePtr(0, _preBuffer->h - (i + offset) + j), newFrame->w * newFrame->format.bytesPerPixel);
+
+				invalidateWindow(false);
+				_vm->yield();
+			}
+
+			newFrame->free();
+			delete newFrame;
+		}
+
+		_rebuildPage = true;
+		invalidateWindow(false);
+	}
+
+	if (_bottom.contains(point) && _curView < _viewCount - 1) {
+		_curView++;
+		_curLineIndex = -1;
+
+		int offset = _vm->_gfx->computeVPushOffset(_vm->getTransitionSpeed());
+
+		// Only draw if transitions are enabled
+		if (offset != 189) {
+			TempCursorChange cursorChange(kCursorWait);
+
+			Graphics::Surface *newFrame = _stillFrames->getFrameCopy(_curView);
+
+			for (int i = 0; i < 189; i += offset) {
+				_preBuffer->move(0, -offset, _preBuffer->h);
+
+				for (int j = 0; j < offset; j++)
+					memcpy(_preBuffer->getBasePtr(0, newFrame->h - offset + j), newFrame->getBasePtr(0, i + j), newFrame->w * newFrame->format.bytesPerPixel);
+
+				invalidateWindow(false);
+				_vm->yield();
+			}
+
+			newFrame->free();
+			delete newFrame;
+		}
+
+		_rebuildPage = true;
+		invalidateWindow(false);
+	}
+
+	if (_putDown.contains(point))
+		((GameUIWindow *)(_parent->getParent()))->_inventoryWindow->destroyBurnedLetterWindow();
+}
+
+void BurnedLetterViewWindow::onMouseMove(const Common::Point &point, uint flags) {
+	_curMousePos = point;
+
+	if (((SceneViewWindow *)_parent)->getGlobalFlags().bcTranslateEnabled == 1) {
+		int lineCount = _viewLineCount[_curView];
+		int textLineNumber = 0;
+		for (int i = 0; i < _curView; i++)
+			textLineNumber += _viewLineCount[i];
+
+		int lineIndex = ((point.y - 2) / (187 / lineCount));
+		if (lineIndex > (lineCount - 1))
+			lineIndex = lineCount - 1;
+
+		if (_curLineIndex != lineIndex) {
+			_curLineIndex = lineIndex;
+			invalidateWindow(false);
+
+			Common::String translatedText = _vm->getString(_translatedTextResourceID + textLineNumber + _curLineIndex);
+			((SceneViewWindow *)_parent)->displayTranslationText(translatedText);
+		}
+
+		return;
+	}
```

Comment:
Using a 'else' at this point would avoid to have a nasty 'return' in the middle of the code